packages/mask/src/plugins/RedPacket/SNSAdaptor/RedPacket/index.tsx

```diff
@@ -134,6 +135,21 @@ export function RedPacket(props: RedPacketProps) {
         else if (canRefund) await refundCallback()
     }, [canClaim, canRefund, claimCallback, refundCallback])
 
+    const onClaimOrRefundOnNative = useCallback(async () => {
+        if (!availability) return
+        nativeAPI?.api.claimOrRefundRedpacket({
+            redpacketPayload: payload,
+            availability: {
+                token_address: availability.token_address,
+                balance: availability.balance,
+                total: availability.total,
+                claimed: availability.claimed,
+                expired: availability.expired
+            },
+            postLink: postLink.toString()
+        })
+    }, [availability, payload, postLink])
```

Comment:
```suggestion
    }, [availability, JSON.stringify(payload), postLink])
```

Reply:
Leave `payload` as a raw object is fine, the native side could easily decode and map it as a JSON object.
With `stringify`, the native side still has to `un-stringify` the payload.

Reply:
`[availability, payload, postLink]` here is a dependency array that `useCallback` uses to compare between two renderings to check if needs to return a new function to assign to `onCalimOrRefundOnNative` here, it's about React not business layer, somehow performance improvement.

In JS, two objects with same shape could referrer to different objects, but serialization of them with `stringify` will be the same. 

Here, if the payload passed from parent component has a same shape among every rendering, and other entities of the dependency array have the same value, `useCallback` will return the exactly same `onClaimOrRefundOnNative`.